rust/migration/src/m20231117_162713_add_created_at.rs

```diff
@@ -0,0 +1,112 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[macro_export]
+macro_rules! static_for {
+    ({ $( $x:expr ),* } as $var:pat_param in $body:block) => {
+        {
+            $(
+                {
+                    let $var = $x;
+                    $body;
+                }
+            )*
+        }
+    };
+}
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        static_for!({
+          (Job::Table, Job::CreatedAt),
+          (OutputDir::Table, OutputDir::CreatedAt),
+          (OutputSymlink::Table, OutputSymlink::CreatedAt),
+          (OutputFile::Table, OutputFile::CreatedAt),
+          (VisibleFile::Table, VisibleFile::CreatedAt),
+          (ApiKey::Table, ApiKey::CreatedAt),
+          (JobUses::Table, JobUses::CreatedAt)
+        } as (t, c) in {
+            manager
+                .alter_table(
+                    Table::alter()
+                        .table(t)
+                        .add_column(
+                            ColumnDef::new(c)
+                                .timestamp()
+                                .not_null()
+                                .default(SimpleExpr::Keyword(Keyword::CurrentTimestamp))
+                        )
+                        .to_owned()
+                ).await?
+        });
+
+        Ok(())
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        static_for!({
+          (Job::Table, Job::CreatedAt),
+          (OutputDir::Table, OutputDir::CreatedAt),
+          (OutputSymlink::Table, OutputSymlink::CreatedAt),
+          (OutputFile::Table, OutputFile::CreatedAt),
+          (VisibleFile::Table, VisibleFile::CreatedAt),
+          (ApiKey::Table, ApiKey::CreatedAt),
+          (JobUses::Table, JobUses::CreatedAt)
+        } as (t, c) in {
+            manager
+                .alter_table(
+                    Table::alter()
+                        .table(t)
+                        .drop_column(c)
+                        .to_owned()
+                ).await?
+        });
+
+        Ok(())
+    }
+}
+
+#[derive(DeriveIden)]
+pub enum Job {
+    Table,
+    CreatedAt,
+}
```

Comment:
I think you can just add the columns to the existing Idens for these tables

Reply:
I would have to import them from the other migration if I did that. And it would also be confusing that the old migration has the `CreatedAt` column but doesn't use it. I think its better to keep migrations self contained

Reply:
My strongest counter here is, what should we do if we (over the series of months) 1) create table Foo 2) delete table Foo 3) create a _new_ table Foo with no association to the original?

Reply:
if that doesn't convenience ya then happy to move forward with that plan

Reply:
That would be a new table that just happens to have the same name but it would be reflected in the fact that it was a different type.